packages/taquito/src/contract/semantic.ts

```diff
@@ -18,6 +19,14 @@ export const smartContractAbstractionSemantic: (p: ContractProvider) => Semantic
       return new BigMapAbstraction(new BigNumber(val.int), schema, provider);
     }
   },
+  sapling_state: (val: MichelsonV1Expression) => {
+    if (!val || !('int' in val) || val.int === undefined) {
+      // Return an empty object in case of missing sapling state ID
+      return {};
```

Comment:
Should we provide an error if id is missing?

Reply:
I don't think we should because this code is executed when we fetch the storage of a contract from the chain and provide an abstraction over it. In fact, this condition should not even happen.. but if it does for any reason, I think it is better to provide an empty sapling state and still return the other values of the storage instead of throwing an exception which the user doesn't have control over. Let me know what you think.

Reply:
Sounds good, thanks for explaining that!